web/modules/weather_data/src/Service/WeatherAlertTrait.php

```diff
@@ -57,27 +57,65 @@ public static function tryParsingDescriptionText($str)
     /**
      * Get active alerts for a WFO grid cell.
      */
-    public function getAlertsForGrid($wfo, $x, $y)
+    public function getAlertsForGrid($wfo, $x, $y, $self = false)
     {
-        $geometry = $this->getGeometryFromGrid($wfo, $x, $y);
-        $point = $geometry[0];
+        if (!$self) {
+            $self = $this;
+        }
 
-        return $this->getAlertsForLatLon($point->lat, $point->lon);
-    }
+        $CACHE_KEY = "alerts $wfo/$x/$y";
+        $cache = $this->cache->get($CACHE_KEY);
```

Comment:
Always use `$this` when using an injected service...

Reply:
Should we just use `$self` in here across the board, or is the idea only to use it to call methods we actively test? (asking because of the call to `$this->cache` etc)